src/client/components/pages/revision.js

```diff
@@ -163,21 +163,21 @@ class RevisionPage extends React.Component {
 		this.handleSubmit = this.handleSubmit.bind(this);
 	}
 
-	handleSubmit(event) {
+	async handleSubmit(event) {
 		event.preventDefault();
 		const data = {
 			note: this.noteInput.value
 		};
-		request.post(`/revision/${this.props.revision.id}/note`)
-			.send(data)
-			.then(() => {
-				location.reload();
-			})
-			.catch((res) => {
-				// TODO: Add proper error handling.
-				const {error} = res.body;
-				return error;
-			});
+		try {
+			await request.post(`/revision/${this.props.revision.id}/note`).send(data);
+			location.reload();
+		}
+		catch (res) {
```

Comment:
A Promise that throws does not throw a response 0bject, so here `res` will not be a response but rather an error object.

You can refer to the superagent documentation for error handling here: https://ladjs.github.io/superagent/#error-handling

Of importance:
> Errors from such responses also contain an err.response field with all of the properties mentioned in "[Response properties](https://ladjs.github.io/superagent/#response-properties)".


I expect `catch(err)` would allow you to access `err.response` and from that access the body, but you'll need to manually check what it returns just to be sure.